server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -320,6 +326,19 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
         // catch early terminated exception and rethrow?
         Bits liveDocs = ctx.reader().getLiveDocs();
         BitSet liveDocsBitSet = getSparseBitSetOrNull(liveDocs);
+        if (isApproximateableRangeQuery()) {
+            ApproximateScoreQuery query = ((ApproximateScoreQuery) ((IndexOrDocValuesQuery) searchContext.query()).getIndexQuery());
+            if (searchContext.size() > 10_000) ((ApproximatePointRangeQuery) query.getApproximationQuery()).setSize(searchContext.size());
+            if (searchContext.request() != null && searchContext.request().source() != null) {
+                FieldSortBuilder primarySortField = FieldSortBuilder.getPrimaryFieldSortOrNull(searchContext.request().source());
+                if (primarySortField != null && primarySortField.missing() == null) {
+                    if (primarySortField.order() == SortOrder.DESC) {
+                        ((ApproximatePointRangeQuery) query.getApproximationQuery()).setSortOrder(SortOrder.DESC);
```

Comment:
This should only kick in if the primary sort field is also the field used for the range query. 

Otherwise, we can't approximate.

Reply:
got it, this makes sense. Essentially does this make sense

```java
if (primarySortField != null 
&& primarySortField.missing() == null 
&& primarySortField.getFieldName().equals(((RangeQueryBuilder) searchContext.request().source().query()).fieldName())
```